pkg/util/glog/glog.go

```diff
@@ -1,101 +1,224 @@
-	if glog.V(glog.Level(level)) {
-		return Log
-	}
-	return None
+func (glogger) Warning(args ...interface{}) {
+	log.Warning(args...)
+}
+
+func (glogger) Errorf(format string, args ...interface{}) {
+	log.Errorf(format, args...)
+}
+
+func (glogger) Error(args ...interface{}) {
+	log.Error(args...)
+}
+
+func (glogger) Fatalf(format string, args ...interface{}) {
+	log.Fatalf(format, args...)
+}
+
+func (glogger) Fatal(args ...interface{}) {
+	log.Fatal(args...)
+}
+
+// gverbose handles glog.V(x) calls
+type gverbose struct {
+	log.Verbose
 }
 
 func (g gverbose) Infof(format string, args ...interface{}) {
 	g.Verbose.Infof(format, args...)
 }
 
-// file logs the provided messages at level or below to the writer, or delegates
+func (g gverbose) Info(args ...interface{}) {
+	g.Verbose.Info(args...)
+}
+
+// FileLogger logs the provided messages at level or below to the writer, or delegates
 // to glog.
-type file struct {
-	w     io.Writer
-	level int
+type FileLogger struct {
+	w     *bufio.Writer
+	level int32
 }
 
-func (f file) Is(level int) bool {
-	return level <= f.level
+// Is returns whether the current logging level is greater than or equal to the parameter
+func (f *FileLogger) Is(level int32) bool {
+	return level <= atomic.LoadInt32(&f.level)
 }
 
-func (f file) V(level int) Logger {
+// V will returns a logger which will discard output if the specified level is greater than the current logging level
+func (f *FileLogger) V(level int32) VerboseLogger {
 	// only log things that glog allows
-	if !glog.V(glog.Level(level)) {
+	if !log.V(log.Level(level)) {
 		return None
 	}
+
 	// send anything above our level to glog
-	if level > f.level {
+	if level > atomic.LoadInt32(&f.level) {
 		return Log
 	}
 	return f
 }
 
-func (f file) Infof(format string, args ...interface{}) {
-	fmt.Fprintf(f.w, format, args...)
-	if !strings.HasSuffix(format, "\n") {
-		fmt.Fprintln(f.w)
+func (f *FileLogger) writeln(line string) {
+	defer f.w.Flush()
+	if atomic.LoadInt32(&f.level) >= 100 {
+		pc := make([]uintptr, 10)
+		runtime.Callers(4, pc)
+		function := runtime.FuncForPC(pc[0])
+		file, line := function.FileLine(pc[0])
+		s := strings.Split(file, "/") // remove absolute path from file
+		file = s[len(s)-1]
+		f.w.WriteString(fmt.Sprintf("[%s:%d] ", file, line-1))
 	}
+
+	f.w.WriteString(line)
+	if !strings.HasSuffix(line, "\n") {
+		f.w.WriteByte('\n')
+	}
+}
+
+func (f *FileLogger) outputf(format string, args ...interface{}) {
+	f.writeln(fmt.Sprintf(format, args...))
+}
+
+func (f *FileLogger) output(args ...interface{}) {
+	f.writeln(fmt.Sprint(args...))
+}
+
+// Infof records an info log entry
+func (f *FileLogger) Infof(format string, args ...interface{}) {
+	f.outputf(format, args...)
+}
+
+// Info records an info log entry
+func (f *FileLogger) Info(args ...interface{}) {
+	f.output(args...)
+}
+
+// Warningf records an warning log entry
+func (f *FileLogger) Warningf(format string, args ...interface{}) {
```

Comment:
If this is going to be a (near) drop-in replacement for glog, it seems like a shame to lose semantic information like glog.Warning versus glog.Error. If we control the logging implementation, we can enforce consistency with or divergence from glog without any difficulty (we can change filtering, output routing, prefixes, add a timestamp/file/line info, etc, based on our internal needs). Delegation to glog seems unnecessary as it appears to actually interfere with the current desired output format.

To a lesser extent, I think developer convenience is something to consider. glog.V(0).Info("Warning..." ) takes more thought than glog.Warning, will lead to inconsistency ("WARNING:", "Warning:", "warning:", or forgetting it altogether) .. and again, valuable meaning on how to handle the message that we might otherwise leverage in the future will be lost. 

It is also worth considering, barring some complex handling of command line flags, glog is made mostly useless when cobra is used in the client program (-v, -vmodules, etc, don't work).